test/prob/test_fixture_ccdf_log.hpp

```diff
@@ -700,6 +700,249 @@ class AgradCcdfLogTestFixture : public ::testing::Test {
     }
   }
 
+  void test_as_scalars_vs_as_vector() {
+    if (!any_vector<T0, T1, T2, T3, T4, T5>::value) {
+      SUCCEED() << "No test for non-vector arguments";
+      return;
+    }
+    vector<double> expected_ccdf_log;
+    vector<vector<double>> parameters;
+    TestClass.valid_values(parameters, expected_ccdf_log);
+
+    vector<double> single_gradients1;
+    vector<double> single_gradients2;
+    vector<double> single_gradients3;
+    vector<var> scalar_vars;
+
+    vector<double> multiple_gradients1;
+    vector<double> multiple_gradients2;
+    vector<double> multiple_gradients3;
+    vector<var> vector_vars;
+
+    T0 p0 = get_params<T0>(parameters, 0);
+    T1 p1 = get_params<T1>(parameters, 1);
+    T2 p2 = get_params<T2>(parameters, 2);
+    T3 p3 = get_params<T3>(parameters, 3);
+    T4 p4 = get_params<T4>(parameters, 4);
+    T5 p5 = get_params<T5>(parameters, 5);
+
+    vector<Scalar0> p0s = {get_param<Scalar0>(parameters[0], 0)};
+    vector<Scalar1> p1s = {get_param<Scalar1>(parameters[0], 1)};
+    vector<Scalar2> p2s = {get_param<Scalar2>(parameters[0], 2)};
+    vector<Scalar3> p3s = {get_param<Scalar3>(parameters[0], 3)};
+    vector<Scalar4> p4s = {get_param<Scalar4>(parameters[0], 4)};
+    vector<Scalar5> p5s = {get_param<Scalar5>(parameters[0], 5)};
+
+    T_return_type single_ccdf_log
+        = TestClass.template ccdf_log<Scalar0, Scalar1, Scalar2, Scalar3,
+                                      Scalar4, Scalar5>(p0s.back(), p1s.back(),
+                                                        p2s.back(), p3s.back(),
+                                                        p4s.back(), p5s.back());
+
+    for (size_t n = 1; n < parameters.size(); n++) {
+      p0s.push_back((is_vector<T0>::value)
+                        ? get_param<Scalar0>(parameters[n], 0)
+                        : p0s.front());
+      p1s.push_back((is_vector<T1>::value)
+                        ? get_param<Scalar1>(parameters[n], 1)
+                        : p1s.front());
+      p2s.push_back((is_vector<T2>::value)
+                        ? get_param<Scalar2>(parameters[n], 2)
+                        : p2s.front());
+      p3s.push_back((is_vector<T3>::value)
+                        ? get_param<Scalar3>(parameters[n], 3)
+                        : p3s.front());
+      p4s.push_back((is_vector<T4>::value)
+                        ? get_param<Scalar4>(parameters[n], 4)
+                        : p4s.front());
+      p5s.push_back((is_vector<T5>::value)
+                        ? get_param<Scalar5>(parameters[n], 5)
+                        : p5s.front());
+
+      single_ccdf_log += TestClass.template ccdf_log<Scalar0, Scalar1, Scalar2,
+                                                     Scalar3, Scalar4, Scalar5>(
```

Comment:
Why templating with scalars, not complete types of the arguments?

Reply:
I'm not sure why they're templated at all. Just copy pasting.

Reply:
I got rid of the template types. Not sure why they were there. Just copy-paste.